File: LocalMaxima.cpp
#include <stdio.h>

void findLocalMaxima(int arr[], int size) {
    if (size < 2) return;
    
    // First element
    if (arr[0] > arr[1]) {
        printf("%d ", arr[0]);
    }
    
    // Middle elements
    for (int i = 1; i < size - 1; i++) {
        if (arr[i] > arr[i - 1] && arr[i] > arr[i + 1]) {
            printf("%d ", arr[i]);
        }
    }
    
    // Last element
    if (arr[size - 1] > arr[size - 2]) {
        printf("%d", arr[size - 1]);
    }
    printf("\n");
}

int main() {
    int arr[] = {2, 5, 7, 8, 4, 3, 5, 9, 4, 6, 8};
    int size = sizeof(arr) / sizeof(arr[0]);
    findLocalMaxima(arr, size);
    return 0;
}

File: LocalMaxima.py
def find_local_maxima(arr):
    n = len(arr)
    
    # First element
    if n > 1 and arr[0] > arr[1]:
        print(arr[0], end=" ")
    
    # Middle elements
    for i in range(1, n - 1):
        if arr[i] > arr[i - 1] and arr[i] > arr[i + 1]:
            print(arr[i], end=" ")
    
    # Last element
    if n > 1 and arr[n - 1] > arr[n - 2]:
        print(arr[n - 1], end=" ")
    print()

arr = [2, 5, 7, 8, 4, 3, 5, 9, 4, 6, 8]
find_local_maxima(arr)
